src/poetry/core/version/pep440/version.py

```diff
@@ -14,6 +16,8 @@
 from poetry.core.version.pep440.segments import ReleaseTag
 
 
+T = TypeVar("T")
```

Comment:
shouldn't this be
```python
T = TypeVar("T", bound="PEP440Version")
```

Reply:
Hm maybe; the current version works as well. I think just `T` is fine because `self` will only ever be a `PEP440Version` or one of it's subclasses. But I can understand if what you suggest is preferred (more obvious). I also haven't tried what happens with `Version.next_major('foobar')` (whether Mypy acepts it because `self` is a `str` and that is compatible with the unbounded type variable `T`)

Reply:
https://mypy.readthedocs.io/en/stable/generics.html#generic-methods-and-generic-self uses the bound